client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js

```diff
@@ -0,0 +1,116 @@
+angular.module('managerApp')
+  .controller('CloudProjectComputeTasksCreateCtrl', class CloudProjectComputeTasksCreateCtrl {
+    constructor(
+      $state, $stateParams, $translate,
+      CloudMessage, CloudProjectCompute, OvhApiCloudProjectInstance, validator,
+      CPC_TASKS,
+    ) {
+      this.$state = $state;
+      this.$stateParams = $stateParams;
+      this.$translate = $translate;
+      this.CloudMessage = CloudMessage;
+      this.CloudProjectCompute = CloudProjectCompute;
+      this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+      this.validator = validator;
+      this.CPC_TASKS = CPC_TASKS;
+    }
+
+    $onInit() {
+      this.serviceName = this.$stateParams.projectId;
+      this.instanceId = this.$stateParams.instanceId;
+      this.instancesAreLoading = true;
+
+      this.model = {
+        action: null,
+        instance: null,
+        schedule: null,
+      };
+
+      this.getRegionsWithWorkflowService();
+    }
+
+    getRegionsWithWorkflowService() {
+      return this.CloudProjectCompute.getRegionsWithWorkflowService(this.serviceName)
+        .then((regions) => {
+          this.regions = regions;
+          return this.getCloudProjectEligibleInstances();
+        })
+        .catch(() => {
+          this.CloudMessage.error(this.$translate.instant('cpc_tasks_resources_error'));
+          this.instancesAreLoading = false;
+        });
+    }
+
+    getCloudProjectEligibleInstances() {
+      return this.CloudProjectCompute.getInstances(this.serviceName)
+        .then((instances) => {
+          this.instances = _.filter(
+            instances,
+            instance => _.includes(this.regions, instance.region),
+          );
+          if (this.instanceId) {
+            this.model.instance = _.find(instances, { id: this.instanceId });
+          }
+        })
+        .catch(() => {
+          this.CloudMessage.error(this.$translate.instant('cpc_tasks_resources_error'));
+        })
+        .finally(() => { this.instancesAreLoading = false; });
+    }
+
+    isActionSelected() {
+      return !_.isNull(this.model.action);
+    }
+
+    isInstanceSelected() {
+      return !_.isNull(this.model.instance);
+    }
+
+    isScheduleSelected() {
+      return !_.isNull(this.model.schedule);
+    }
+
+    createTask() {
+      if (this.model.action.id === 'snapshot') {
```

Comment:
You will add a `if` for each kind of task ? You could call the function according to a convention instead
```js
this[`generate${kindOfTask}TaskParams`]();
```